test/vfd_swmr_gfail_writer.c

```diff
@@ -0,0 +1,798 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Copyright by The HDF Group.                                               *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of HDF5.  The full HDF5 copyright notice, including     *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the root of the source code       *
+ * distribution tree, or in https://www.hdfgroup.org/licenses.               *
+ * If you do not have access to either file, you may request a copy from     *
+ * help@hdfgroup.org.                                                        *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+/*
+ *  Purpose: To demostrate the four issues uncovered during the test plan
+ *           1.4 part 1.
+ *
+ *           Setup:
+ *           Writer:
+ *            1) Create an HDF5 file
+ *            2) Create many groups
+ *            3) Use the named pipe to send the reader a message to start verifying
+ *            4) Call H5Fvfd_swmr_end_tick immeidately after sending out the message to the reader
+ *            5) Sleep for two ticks before receiving a message from the reader that informs the reader
+ *               started verifying
+ *            6) If the option to delete the group is off(by default), just sleep for a relatively long time,
+ *                  then close the HDF5 file.
+ *               Else delete the last 1000 groups that are just created if the total number of
+ *                  created groups is greater than 1000.
+ *                  Call H5Fvfd_swmr_end_tick sleep for a relatively long time,
```

Comment:
Just wondering why is H5Fvfd_swmr_end_tick needed here?

Reply:
Actually I listed a comment just above using the code, say " /* end tick,may be not necessary. */"
So it may not be needed when groups are deleted.  I used it here to just make the reader see the metadata as soon as possible. The tests may take a little bit shorter. We will see what John said about it.

  ```
for (step = s.nsteps - 1; step >= (s.nsteps - 1000); step--) {
                dbgf(2, "writer: deleting step %d\n", step);
    
                wg_ret = delete_group(&s, step);
                if (wg_ret == false) {
                    HDprintf("delete group_operations failed\n");
                    TEST_ERROR;
                }
            }
            /* end tick,may be not necessary. */
            if (H5Fvfd_swmr_end_tick(s.file) < 0)
                TEST_ERROR;
```